### PR TITLE
fix ModelToTensorBoard max_size = -1

### PR DESCRIPTION
**Pull request by sam-willis**
**PR type:** bugfix

## Summary

Currently `ModelToTensorBoard` says in it's initialiser that passing -1 to max size will track all the parameters of a model, but currently this is not respected.

### Minimal working example

```python
# Put your example code in here
```

## PR checklist

- [ ] New features: code is well-documented
  - [ ] detailed docstrings (API documentation)
  - [ ] notebook examples (usage demonstration)
- [ ] The bug case / new feature is covered by unit tests
- [ ] Code has type annotations
- [ ] I ran the black+isort formatter (`make format`)
- [ ] I locally tested that the tests pass (`make check-all`)

### Release notes

**Fully backwards compatible:** yes

**If not, why is it worth breaking backwards compatibility:**

**Commit message (for release notes):**

* ...

### CODE REVIEW

**Comment by sam-willis** [comment]
The failing test is `test_scipy_jit`, this seem unrelated to the change I've made in this PR.

File: gpflow/monitor/tensorboard.py [hunk 1]
```diff
@@ -117,11 +117,9 @@ def _summarize_parameter(self, name: str, param: Union[Parameter, tf.Variable]):
...
             )
 
-        if size == 1:
-            tf.summary.scalar(name, param[0], step=self.current_step)
-        else:
-            for i in range(min(size, self.max_size)):
-                tf.summary.scalar(f"{name}[{i}]", param[i], step=self.current_step)
+        it = range(size) if self.max_size == -1 else range(min(size, self.max_size))
+        for i in it:
+            tf.summary.scalar(f"{name}[{i}]", param[i], step=self.current_step)
```
Comment (st--) [inline gpflow/monitor/tensorboard.py:122]: This changes behaviour (namely, it'll always add the `[0]` for scalar parameters). Might be nice to only have your change in the else: branch of the original code?

**Comment by st--** [comment]
Hi @sam-willis , thanks for spotting & fixing this! The unit tests will be fixed by #1625. Would you be happy to address the one comment I left? We can merge it once the other PR is in.

File: gpflow/monitor/tensorboard.py (hunk 1 above)
Comment (sam-willis) [reply gpflow/monitor/tensorboard.py:122]: Ah, makes sense! Sure I'll make the change

**Review by st-- — APPROVED** [review]
looks good to me, thanks for the fix. (just wondering if it's worth bothering to add a regression test for this?)

**Comment by sam-willis** [comment]
It probably would, but I don't have time to do it at the moment, it's up to you whether you'd rather merge now, or leave it with me in which case it'll get the test when I'm next at a loose end.

**Pull request merged**